hydra-node/test/Hydra/HeadLogicSpec.hs

```diff
@@ -722,6 +724,117 @@ spec =
         update bobEnv ledger (inClosedState threeParties) collectOtherHead
           `shouldBe` Error (NotOurHead{ourHeadId = testHeadId, otherHeadId})
 
+      describe "SideLoad InitialSnapshot" $ do
+        it "accept side load initial snapshot with idempotence" $ do
+          let s0 = inOpenState threeParties
+              initialSn = InitialSnapshot testHeadId mempty
+              snapshot0 = getSnapshot initialSn
+          getConfirmedSnapshot s0 `shouldBe` Just snapshot0
+          sideLoadedState <- runHeadLogic bobEnv ledger s0 $ do
+            step $ ClientInput (SideLoadSnapshot initialSn)
+            getState
+          getConfirmedSnapshot sideLoadedState `shouldBe` Just snapshot0
+
+        it "reject side load wrong initial snapshot" $ do
+          let s0 = inOpenState threeParties
+              initialSn = InitialSnapshot testHeadId mempty
+              snapshot0 = getSnapshot initialSn
+          getConfirmedSnapshot s0 `shouldBe` Just snapshot0
+          let wrongInitialSnapshot = InitialSnapshot testHeadId (utxoRef 2)
+          update bobEnv ledger s0 (ClientInput (SideLoadSnapshot wrongInitialSnapshot))
+            `shouldBe` Error (AssertionFailed "InitalSnapshot side loaded does not match last known.")
+          getConfirmedSnapshot s0 `shouldBe` Just snapshot0
+
+        prop "ignores side load initial snapshot of another head" $ \otherHeadId -> do
+          let s0 = inOpenState threeParties
+              initialSn = InitialSnapshot testHeadId mempty
+              snapshot0 = getSnapshot initialSn
+          getConfirmedSnapshot s0 `shouldBe` Just snapshot0
+          let initialSnapshotOtherHead = InitialSnapshot otherHeadId mempty
+          update bobEnv ledger s0 (ClientInput (SideLoadSnapshot initialSnapshotOtherHead))
+            `shouldBe` Error (NotOurHead{ourHeadId = testHeadId, otherHeadId})
+          getConfirmedSnapshot s0 `shouldBe` Just snapshot0
+
+      describe "SideLoad ConfirmedSnapshot" $ do
+        -- Given a list of transactions each depending on the previous.
+        let tx1 = SimpleTx 1 mempty (utxoRef 2) -- No inputs, requires no specific starting state
+            tx2 = SimpleTx 2 (utxoRef 2) (utxoRef 3)
+            tx3 = SimpleTx 3 (utxoRef 3) (utxoRef 4)
+            snapshot1 = Snapshot testHeadId 0 1 [tx1] (utxoRef 2) Nothing Nothing
```

Comment:
you could test also how  `utxoToDe/Commit` is behaving - related to your review comment in the head logic.